pywinauto/unittests/test_uiawrapper.py

```diff
@@ -73,15 +73,17 @@ def tearDown(self):
         def test_issue_296(self):
             """Test handling of disappered descendants"""
             wrp = self.dlg.wrapper_object()
-            orig = wrp.element_info._element.FindAll
-            wrp.element_info._element.FindAll = mock.Mock(side_effect=ValueError("Mocked value error"),
-                                                          return_value=[])  # empty list
+            orig = uia_defs.IUIA().raw_tree_walker.GetFirstChildElement
+            uia_defs.IUIA().raw_tree_walker.GetFirstChildElement = mock.Mock(
+                side_effect=ValueError("Mocked value error"),
+                return_value=[])  # empty list
             self.assertEqual([], wrp.descendants())
             exception_err = comtypes.COMError(-2147220991, "Mocked COM error", ())
-            wrp.element_info._element.FindAll = mock.Mock(side_effect=exception_err,
-                                                          return_value=[])  # empty list
+            uia_defs.IUIA().raw_tree_walker.GetFirstChildElement = mock.Mock(
+                side_effect=exception_err,
+                return_value=[])  # empty list
             self.assertEqual([], wrp.descendants())
-            wrp.element_info._element.FindAll = orig  # restore the original method
+            uia_defs.IUIA().raw_tree_walker.GetFirstChildElement = orig  # restore the original method
```

Comment:
Honestly, we should switch to a mock context manager, because the original method won't be restored if test asserts and it may affect outcome of other tests.

Reply:
Hmm, I tried to write this:
```
wrp = self.dlg.wrapper_object()
with mock.patch.object(uia_defs.IUIA().raw_tree_walker, 'GetFirstChildElement') as mock_get_first_child:
    mock_get_first_child = mock.Mock(
        side_effect=ValueError("Mocked value error"),
        return_value=[])  # empty list
    self.assertEqual([], wrp.descendants())
```
and got exception on `isinstance` check, caused by replacement to `MagicMock` object:
```
File "C:\repo\pywinauto\github\pywinauto\pywinauto\windows\uia_element_info.py", line 337, in iter_descendants
    yield UIAElementInfo(child, cache_enable)
File "C:\repo\pywinauto\github\pywinauto\pywinauto\windows\uia_element_info.py", line 113, in __init__
    "with integer or IUIAutomationElement instance only!")
TypeError: UIAElementInfo object can be initialized with integer or IUIAutomationElement instance only!
```

I will try to find a solution to this problem.

Reply:
done at https://github.com/pywinauto/pywinauto/pull/1001/commits/99a2b300ac5675d305222b9aa09b7c709db735dc